src/node_file.cc

```diff
@@ -2801,62 +2787,51 @@ static bool FileURLToPath(
 
   // a..z A..Z
   if (letter < 'a' || letter > 'z' || sep != ':') {
-    env->isolate()->ThrowException(ERR_INVALID_FILE_URL_PATH(
-        env->isolate(), "File URL path must be absolute"));
-
+    THROW_ERR_INVALID_FILE_URL_PATH(env, "File URL path must be absolute");
     return false;
   }
 
-  result_file_path = decoded_pathname.substr(1);
+  *result_file_path = decoded_pathname.substr(1);
 
   return true;
 #else   // _WIN32
-  std::string_view hostname = file_url.get_hostname();
-
-  if (hostname.size() > 0) {
+  if (!file_url.has_empty_hostname()) {
     std::string error_message =
-        std::string("File URL host must be \"localhost\" or empty on ") +
+        "File URL host must be \"localhost\" or empty on " +
         std::string(per_process::metadata.platform);
-    env->isolate()->ThrowException(
-        ERR_INVALID_FILE_URL_HOST(env->isolate(), error_message.c_str()));
-
+    THROW_ERR_INVALID_FILE_URL_HOST(env, error_message.c_str());
```

Comment:
ditto (untested code)
```suggestion
    THROW_ERR_INVALID_FILE_URL_HOST(env, "File URL host must be \"localhost\" or empty on %s", per_process::metadata.platform);
```
(and drop the error_message assignment above)